includes/admin/class-wcs-admin-meta-boxes.php

```diff
@@ -85,8 +84,10 @@ public function add_meta_boxes() {
 		add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', 'shop_subscription', 'normal', 'low' );
 
 		// Only display the meta box if an order relates to a subscription
-		if ( 'shop_order' === WC_Data_Store::load( 'order' )->get_order_type( $post_ID ) && wcs_order_contains_subscription( $post_ID, 'any' ) ) {
-			add_meta_box( 'subscription_renewal_orders', __( 'Related Orders', 'woocommerce-subscriptions' ), 'WCS_Meta_Box_Related_Orders::output', 'shop_order', 'normal', 'low' );
+		if ( wcs_order_contains_subscription( $post_or_order_object, 'any' ) ) {
+			// If HPOS is enabled, we need to get the screen id and provide it to add_meta_box().
+			$screen = wcs_is_custom_order_tables_usage_enabled() ? wc_get_page_screen_id( 'shop-order' ) : 'shop_order';
```

Comment:
Added this `$current_screen->id === $screen` condition, as it was rendering on other screens (e.g. Edit Product).